app/controllers/users_controller.rb

```diff
@@ -1,7 +1,60 @@
 class UsersController < ApplicationController
 
   def list
-    @users = User.all
+    # Display list based on action_type param
+    conditions = {}
+    if params[:action_type] == 'trim'
+      conditions[:trimmed] = true
+    elsif params[:action_type] == 'massage_and_trim'
+      conditions[:massaged] = true
+      conditions[:trimmed] = true
+    elsif params[:action_type] == 'shave'
+      conditions[:shaved] = true
+    end
+    @users = User.where(conditions)
   end
   alias :index :list
+
+  # Set trimmed to user
+  # GET /users/:id/trim
```

Comment:
In general, it is very poor form on the web to use a GET as something which DOES an action.  You should make your actions be either a PUT (update) or POST (create).  All browsers will warn the user to redo the action should they hit "refresh" on their browser.

Also, this is not very RESTful.  You will find that your code becomes much simpler if you use the rails routing and resources to rewrite these actions restfully.